--- git status ---
HEAD detached from 5c3459e
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   deploy/config.py
	modified:   source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/custom_rewards.py
	modified:   source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/locomotion/go2_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	concurrent_state_estimator.pth
	tested_policies/go2/2026-07-09_06-46-39/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/deploy/config.py b/deploy/config.py
index 30fa0ef..bd730d4 100644
--- a/deploy/config.py
+++ b/deploy/config.py
@@ -24,7 +24,7 @@ elif(robot == "go2"):
     Kp_stand_up_and_down = 50.
     Kd_stand_up_and_down = 5.
 
-    policy_folder_path = dir_path + "/../tested_policies/" + robot + "/concurrent_symm"
+    policy_folder_path = dir_path + "/../tested_policies/" + robot + "/2026-07-09_06-46-39"
 
 elif(robot == "b2"):
     Kp_walking = 100.
diff --git a/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/custom_rewards.py b/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/custom_rewards.py
index 36f8e56..6aa6f5b 100644
--- a/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/custom_rewards.py
+++ b/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/custom_rewards.py
@@ -55,7 +55,7 @@ def track_height_exp(self) -> torch.Tensor:
 
 def track_lin_vel_xy_exp(self) -> torch.Tensor:
     lin_vel_error = torch.sum(torch.square(self._commands[:, :2] - self._robot.data.root_lin_vel_b[:, :2]), dim=1)
-    lin_vel_error_mapped = torch.exp(-lin_vel_error / 0.25)
+    lin_vel_error_mapped = torch.exp(-lin_vel_error / 0.1)
     return lin_vel_error_mapped
 
 
@@ -107,7 +107,7 @@ def track_ang_vel_xy_l2(self) -> torch.Tensor:
 
 def track_ang_vel_z_exp(self) -> torch.Tensor:
     yaw_rate_error = torch.square(self._commands[:, 2] - self._robot.data.root_ang_vel_b[:, 2])
-    yaw_rate_error_mapped = torch.exp(-yaw_rate_error / 0.25)
+    yaw_rate_error_mapped = torch.exp(-yaw_rate_error / 0.1)
     return yaw_rate_error_mapped
 
 
@@ -178,17 +178,54 @@ def joints_energy_l1(self) -> torch.Tensor:
 
 
 def feet_air_time(self) -> torch.Tensor:
-    mode_time = 0.5
-    current_air_time = self._contact_sensor.data.current_air_time[:, self._feet_contact_sensor_ids]
-    current_contact_time = self._contact_sensor.data.current_contact_time[:, self._feet_contact_sensor_ids]
-    t_max = torch.max(current_air_time, current_contact_time)
-    t_min = torch.clip(t_max, max=mode_time)
-    feet_air_time_per_leg = torch.where(t_max < mode_time, t_min, torch.zeros_like(t_min))
-    feet_air_time_per_leg -= torch.where(
-        current_air_time > mode_time, current_air_time - mode_time, torch.zeros_like(current_air_time)
+    desired_contact_time = 0.47
+    desired_air_time = 0.25
+
+    current_air_time = self._contact_sensor.data.current_air_time[
+        :, self._feet_contact_sensor_ids
+    ]
+    current_contact_time = self._contact_sensor.data.current_contact_time[
+        :, self._feet_contact_sensor_ids
+    ]
+
+    in_contact = current_contact_time > 0.0
+
+    current_time = torch.where(
+        in_contact,
+        current_contact_time,
+        current_air_time,
+    )
+
+    desired_time = torch.where(
+        in_contact,
+        torch.full_like(current_time, desired_contact_time),
+        torch.full_like(current_time, desired_air_time),
+    )
+
+    # From 0 to 1 until reach the target
+    bounded_reward = torch.clamp(
+        current_time / desired_time,
+        max=1.0,
     )
-    feet_air_time = torch.sum(feet_air_time_per_leg, dim=1) * (torch.norm(self._commands[:, :3], dim=1) > 0.1)
-    return feet_air_time
+
+    # After reaching the target, apply a penalty for exceeding the desired time
+    excess_penalty = torch.clamp(
+        (current_time - desired_time) / desired_time,
+        min=0.0,
+    )
+
+    feet_reward_per_leg = bounded_reward - excess_penalty
+
+    # Limite inferiore opzionale per evitare penalità enormi.
+    feet_reward_per_leg = torch.clamp(
+        feet_reward_per_leg,
+        min=-1.0,
+        max=1.0,
+    )
+
+    should_move = torch.norm(self._commands[:, :3], dim=1) > 0.01
+
+    return torch.sum(feet_reward_per_leg, dim=1) * should_move
 
 
 def feet_air_time_variance(self):
diff --git a/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/locomotion/go2_env_cfg.py b/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/locomotion/go2_env_cfg.py
index 2383e1e..ebbef43 100644
--- a/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/locomotion/go2_env_cfg.py
+++ b/source/basic_locomotion_isaaclab/basic_locomotion_isaaclab/tasks/locomotion/go2_env_cfg.py
@@ -133,7 +133,7 @@ class Go2FlatEnvCfg(DirectRLEnvCfg):
     observation_space += 12 # joint velocities
     observation_space += 12 # last actions
 
-    use_clock_signal = True
+    use_clock_signal = False
     if(use_clock_signal):
         observation_space += 4 # clock signal for periodic gait
 
@@ -339,16 +339,16 @@ class Go2FlatEnvCfg(DirectRLEnvCfg):
 
 
     # Feet reward scale
-    feet_air_time_reward_scale = 0.5 * 0.0
-    feet_air_time_variance_reward_scale = -1.0 * 0.0
+    feet_air_time_reward_scale = 0.25
+    feet_air_time_variance_reward_scale = -1.0
 
     feet_height_clearance_aperiodic_reward_scale = 0.25 * 0.0  
     feet_height_clearance_periodic_reward_scale = 0.25 * 0.0
     
-    feet_height_clearance_mujoco_aperiodic_reward_scale = 0.25 * 0.0
-    feet_height_clearance_mujoco_periodic_reward_scale = 0.25# * 0.0
+    feet_height_clearance_mujoco_aperiodic_reward_scale = 0.25
+    feet_height_clearance_mujoco_periodic_reward_scale = 0.25 * 0.0
     
-    feet_slide_reward_scale = -0.25 * 0.0
+    feet_slide_reward_scale = -0.25
     
     feet_to_hip_distance_reward_scale = 1.5
     # This is used in loocmotion_env.py for the above reward
@@ -364,13 +364,13 @@ class Go2FlatEnvCfg(DirectRLEnvCfg):
 
 
     # Contact suggestion reward scale
-    periodic_contact_suggestion_reward_scale =  0.5
+    periodic_contact_suggestion_reward_scale =  0.5 * 0.0
     # Desired step freq and duty factor (if periodic gait contact suggestion is used)
     desired_step_freq = 1.4
     desired_duty_factor = 0.65
     desired_phase_offset = [0.0, 0.5, 0.5, 0.0] #FL, FR, RL, RR
 
-    stance_contact_suggestion_reward_scale = 0.25
+    stance_contact_suggestion_reward_scale = 1.0
 
 
 